File: Go/백준/Platinum/25196. 숲속에서 새 구경하기/숲속에서 새 구경하기.go
package main

import (
	"bufio"
	"math"
	"os"
	"strconv"
)

func max(args ...int) int {
	res := 0
	for _, arg := range args {
		if res < arg {
			res = arg
		}
	}
	return res
}

func min(args ...int) int {
	res := math.MaxInt
	for _, arg := range args {
		if res > arg {
			res = arg
		}
	}
	return res
}

func main() {
	bs.Split(bufio.ScanWords)
	defer bw.Flush()

	Av, As, Ae := intInput(), intInput(), intInput()
	Bv, Bs, Be := intInput(), intInput(), intInput()
	Cv, Cs, Ce := intInput(), intInput(), intInput()

	limit := Av * Bv * Cv
	for Ae <= limit && Be <= limit && Ce <= limit {
		s := max(As, Bs, Cs)
		e := min(Ae, Be, Ce)

		if s <= e {
			bw.WriteString(strconv.Itoa(s))
			return
		}

		if Ae <= Be && Ae <= Ce {
			As += Av
			Ae += Av
		} else if Be <= Ae && Be <= Ce {
			Bs += Bv
			Be += Bv
		} else {
			Cs += Cv
			Ce += Cv
		}
	}
	bw.WriteString("-1")
}

var (
	bs = bufio.NewScanner(os.Stdin)
	bw = bufio.NewWriter(os.Stdout)
)

func input() string {
	bs.Scan()
	return bs.Text()
}

func intInput() int {
	num, _ := strconv.Atoi(input())
	return num
}

File: PyPy3/백준/Platinum/25196. 숲속에서 새 구경하기/숲속에서 새 구경하기.py
Av,As,Ae = map(int,input().split())
Bv,Bs,Be = map(int,input().split())
Cv,Cs,Ce = map(int,input().split())

limit = Av * Bv * Cv
while Ae <= limit and Be <= limit and Ce <= limit:
    s = max(As,Bs,Cs)
    e = min(Ae,Be,Ce)

    if s <= e:
        exit(print(s))
    if Ae <= Be and Ae <= Ce:
        As += Av
        Ae += Av
    elif Be <= Ae and Be <= Ce:
        Bs += Bv
        Be += Bv
    else:
        Cs += Cv
        Ce += Cv
print(-1)
